File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);

        int left = 0;
        int right = people.length - 1;
        int answer = 0;

        while (left <= right) {
            if (people[left] + people[right] <= limit) {
                left++; // 가벼운 사람도 같이 탐
            }
            right--; // 무거운 사람은 항상 탐
            answer++;
        }

        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    
    people.sort()
    left, right = 0, len(people) - 1
    
    # 가장 가벼운 사람과 가장 무거운 사람을 묶어서 태움
    while left < right:
        if people[left] + people[right] <= limit:
            answer += 1
            left += 1
            right -= 1
        else:
            answer += 1
            right -= 1
            
    if left == right:
        answer += 1
        
    return answer